react-renderer/src/inspector/views/mod/AttrPicker.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+import LabeledInput from "./LabeledInput"
+
+interface IProps {
+    shape: IShape;
+    sAttrs: IShapeDef;
+    modAttr(attrname: string, attrval: Value<any>): void;
+}
+
+interface IShapeDef {
+    shapeType: string;
+    properties: object;
+}
+
+// Q - should this update shape properties in state? not really necessary functionally but maybe ideologically
```

Comment:
Does this not update shape properties in the state? I expected it to do so, but if not can you say more about how what it is updating?

Reply:
^Possibly related: my comments in `Mod.tsx` (github isn't letting me link them directly)

Reply:
@hypotext It doesn't change the state of that component specifically, just passes up the changes to `App.tsx` where that state gets modified (which then triggers a re-render, so it does change the state indirectly...)